lib/mvi/messages/message_builder.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+require 'ox'
+
+module MVI
+  module Messages
+    module MessageBuilder
+      def doc
+        @doc ||= Ox::Document.new(version: '1.0')
+      end
+
+      def header(extension)
+        @message = idm(extension)
```

Comment:
Does this create the instance variable within the scope of the module and potentially keep the existing `@message` object around for subsequent requests and cause potential threading issues?